Replace size_t format string in InfillTest.cpp

onitake opened the pull request:
# Description

<!-- Please include a summary of which issue is fixed or feature was added. Please also include relevant motivation and context. 
If this pull request adds settings definitions for machines/materials, list them here. 

This fixes... OR This improves... -->
The format string in tests/InfillTest.cpp is not correct for the (last) `size_t` argument, causing test failure on 32-bit architectures.
This PR replaces the `%lld` with the correct C99 format string for `size_t`: `%zu`.

It's expected that this will work with every current C standard library used for CuraEngine.
If this is not the case and C90 C library compatibility is desired, a more conservative format string can be used, such as `%llu`. This was tested successfully on i686 with gcc 12/glibc 2.37, even though it's not 100% correct for a 32-bit `size_t`.

Fixes: #1897 

## Type of change

<!-- Please delete options that are not relevant. -->

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

<!-- Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration -->

- [ ] Build on amd64 and i686 Linux
- [ ] Run unit test InfillTest

**Test Configuration**:
* Operating System: Debian trixie/sid
* amd64 (native) AND i686 (chroot)

# Checklist:

- [x] My code follows the style guidelines of this project as described in [UltiMaker Meta](https://github.com/Ultimaker/Meta)
- [x] I have read the [Contribution guide](https://github.com/Ultimaker/Cura/blob/main/contributing.md)
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have uploaded any files required to test this change

jellespijker commented on tests/InfillTest.cpp:
```diff
@@ -100,7 +100,7 @@ class InfillTestParameters
     {
         // FIXME: Once we are using spdlog as logger, we'll also use fmt::format() here, see CURA-8258.
-        name = makeName("InfillTestParameters_P%d_Z%d_C%d_L%lld__%lld", static_cast<int>(params.pattern), static_cast<int>(params.zig_zagify), static_cast<int>(params.connect_polygons), params.line_distance, test_polygon_id);
+        name = makeName("InfillTestParameters_P%d_Z%d_C%d_L%lld__%zu", static_cast<int>(params.pattern), static_cast<int>(params.zig_zagify), static_cast<int>(params.connect_polygons), params.line_distance, test_polygon_id);
```
```suggestion
        name = makeName(fmt::format("InfillTestParameters_P{}_Z{}_C{}_L{}__{}", static_cast<int>(params.pattern), static_cast<int>(params.zig_zagify), static_cast<int>(params.connect_polygons), params.line_distance, test_polygon_id));
```
An untested suggestion which would require `#include <fmt/format.h>` and the additional removal of the now redudant TODO on L102, also need to double check if this requires specific formatters

jellespijker reviewed (changes requested):
Hi @onitake I'm just curious. I noticed you asking some specific Linux questions libCharon now a PR for 32bit hardware. I'm guessing you're a package maintainer for one of the distro's.

We recently added fmt https://fmt.dev/latest/index.html as one of the dependencies. Since the TODO above states that this should be changed anyhow once we use fmt I would prefer it if this PR uses the fmt lib.

onitake commented:
> Hi @onitake I'm just curious. I noticed you asking some specific Linux questions libCharon now a PR for 32bit hardware. I'm guessing you're a package maintainer for one of the distro's.

That is correct, I'm one of the maintainers of the Debian packages: https://tracker.debian.org/pkg/cura-engine
Maintenance got a bit stuck lately due to lack of time and the switch from CMake to Conan, but we're intent to get at least 5.0 maintained in vanilla Debian.

Since Debian still includes some 32-bit release architectures, we need to ensure compatibility if possible.

> We recently added fmt https://fmt.dev/latest/index.html as one of the dependencies. Since the TODO above states that this should be changed anyhow once we use fmt I would prefer it if this PR uses the fmt lib.

Ok, I can look into that. I don't know that fmt library, but it doesn't sounds straight-forward.

jellespijker commented:
basically the same syntax as the Python format strings

onitake commented:
Ok, so I replaced `makeName` with `fmt::format`.
The format string explicitly uses decimal format `{:d}`, so I dropped the `static_cast`s - except for the `EFillMethod` type.

It looks like fmtlib won't automatically cast enum classes to int starting from v7, to be in line with C++20 semantics.
You either need to add a custom formatter (as in https://fmt.dev/latest/api.html#formatting-user-defined-types) or cast it to the underlying type yourself. See https://github.com/fmtlib/fmt/issues/2704 for an example and the reasoning.

jellespijker reviewed (approved):
LGTM;

Thank you for you contribution and finishing my todo.
